src/compile/data/optimize.ts

```diff
@@ -195,7 +202,7 @@ function optimizationDataflowHelper(dataComponent: DataComponent) {
  * Optimizes the dataflow of the passed in data component.
  */
 export function optimizeDataflow(data: DataComponent) {
-  for (let i = 0; i < 5; i++) {
+  for (let i = 0; i < MAX_OPTIMIZATION_RUNS; i++) {
```

Comment:
Let's add a warning if we reach the `MAX_OPTIMIZATION_RUNS`. I'd like to know when this happens.